ndk/src/main/jni/utils/string.h

```diff
@@ -27,6 +28,12 @@ void bsg_strncpy(char *dst, char *src, size_t len) __asyncsafe;
  * Copy a string from src to dst, null padding the rest
  */
 void bsg_strncpy_safe(char *dst, char *src, int dst_size);
+
+/**
+ * True if first and second are the same length and contain the same characters
+ * in the same order.
+ */
+bool bsg_strequal(char *first, char *second) __asyncsafe;
```

Comment:
I can't find any usage of this function in the project. We should remove it if it's not used here, or add it in a separate PR if it's needed more generally.